cuckoo/cuckoo_result.py

```diff
@@ -1737,8 +1797,11 @@ def _filter_process_tree_against_safe_hashes(process_tree: List[Dict[str, Any]],
     :param task_id: An integer representing the Cuckoo Task ID
     :return: A list of processes in a tree structure, with the safe branches filtered out
     """
-    process_tree_hashes = _create_process_tree_hashes(process_tree)
     stripped_process_tree = _strip_process_tree(process_tree)
+    process_tree_hashes = _create_process_tree_hashes(stripped_process_tree)
+    leaf_hashes = _create_leaf_hashes(stripped_process_tree)
+    stripped_root_leaf_paths = _create_root_leaf_paths(stripped_process_tree)
+    _write_tree_and_leaf_hashes(stripped_root_leaf_paths, leaf_hashes, task_id)
     _write_tree_and_hashes(stripped_process_tree, process_tree_hashes, task_id)
     _remove_safe_roots(process_tree, process_tree_hashes, safe_process_tree_hashes)
```

Comment:
Can we have a function that removes safe leaves as well?

Reply:
Added